micropython/drivers/imu/bmm150/bmm150.py

```diff
@@ -0,0 +1,188 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2023 Arduino SA
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+Basic example usage:
+
+import time
+from bmm150 import BMM150
+from machine import Pin, SPI, I2C
+
+# Init in I2C mode.
+imu = BMM150(I2C(1, scl=Pin(15), sda=Pin(14)))
+
+# Or init in SPI mode.
+# TODO: Not supported yet.
+# imu = BMM150(SPI(5), cs_pin=Pin(10))
+
+while (True):
+    print('magnetometer: x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*imu.magnet()))
+    time.sleep_ms(100)
+"""
+
+import array
+import time
+from micropython import const
+
+_DEFAULT_ADDR = const(0x10)
+_CHIP_ID = const(0x40)
+_DATA = const(0x42)
+_POWER = const(0x4B)
+_OPMODE = const(0x4C)
+_INT_STATUS = const(0x4A)
+_TRIM_X1 = const(0x5D)
+_TRIM_Y1 = const(0x5E)
+_TRIM_Z4_LSB = const(0x62)
+_TRIM_Z2_LSB = const(0x68)
+_XYAXES_FLIP = const(-4096)
+_ZHAXES_FLIP = const(-16384)
+
+
+class BMM150:
+    def __init__(
+        self,
+        bus,
+        cs_pin=None,
```

Comment:
I know `lsm6dsox` uses `cs_pin`, but other drivers like `ssd1360`, `nrf25l01` and `sdcard` all use `cs`, and initialising things like `machine.I2C`, `machine.SPI` and `machine.UART` all use the name of the pin (eg `scl`) without a `_pin` suffix.  So I think we should try to match that standard of using the pin name without any suffix, and just call this parameter `cs`.

The more consistent we make this stuff now, the better it will be for users in the long run.

Reply:
@dpgeorge Should I change the `lsm6dsox` too, it did go through some refactoring, or just the `bmm150` ?

Reply:
> Should I change the lsm6dsox too

Yes please.  Since some of its methods were renamed, may as well change the `cs` argument as well.